### PR TITLE
tribler: 7.1.2 -> 7.4.0-exp1 (python 3)

### PR DESCRIPTION
**Pull request by d-goldin**
###### Motivation for this change

Updating version for tribler to use python 3.
One most notable change is dropping the optional `pythonPackages.meliae`
as it does not support python 3 and addition pony, lz4 and pyqtgraph.

Depends on: https://github.com/NixOS/nixpkgs/pull/74193
Related: https://github.com/NixOS/nixpkgs/pull/73927

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @xvapx

### CODE REVIEW

**Comment by d-goldin** [comment]
Some info about this particular version, since it has a funky name:

> Tribler 7.4.0-exp1 is the Python3 version for stable 7.3.2 release. It includes numerous bug fixes and Python3 fixes over 7.3.2 without any significant feature update.
From https://github.com/Tribler/tribler/releases

File: pkgs/applications/networking/p2p/tribler/default.nix
Comment (flokli) [inline pkgs/applications/networking/p2p/tribler/default.nix]: `wrapPython` and `makeWrapper` should be `nativeBuildInputs` - not sure about imagemagick - where is that required?

Comment (d-goldin) [reply pkgs/applications/networking/p2p/tribler/default.nix]: Right. I'll double check the whole expression and deps - I was mostly focused on getting python 3 to work.

Comment (flokli) [reply pkgs/applications/networking/p2p/tribler/default.nix]: Thanks :-)

Comment (d-goldin) [reply pkgs/applications/networking/p2p/tribler/default.nix]: imagemagick: was used to strip ICC profiles from UI graphics that used to cause issues: https://github.com/Tribler/tribler/pull/1561. I ditched that now and also moved those into nativeBuildInputs.

**Comment by d-goldin** [comment]
A few remarks:

* Ditched a few dependencies that aren't used (upstream doc is also very conflicting about that, with various out-dated files here and there and various sources)
* There is a BTC functionality, which depends on some unclear version of `bitcoinlib` (uses `import bitcoinlib`) that does not seem to be the one from pypi. So this feature doesn't work.
* The video playback feature doesn't work for me
* The trust network view throws an error;
```
  File "/nix/store/mn14ajx3sv6vfi8dihdfkpl8jm23m42i-tribler-7.4.0-exp1/Tribler/Core/Modules/restapi/trustview_endpoint.py", line 57, in get_node
    return self.node[self.peers.index(peer_key)]
AttributeError: 'TrustGraph' object has no attribute 'node'
```

But overall it manages to do some search, discovery and torrent downloading. I'd say it's in medium shape.

 Keeping commits separate for legibility. Will squash once maintainer and reviewers are happy.

**Review by xvapx — APPROVED** [review]
Looks good to me.
The bitcoin feature is new so it's not a regression for it to be broken.
The video playback is a regression, but I think it's worth it to upgrade to python3, I'll look into it after this is merged and try to fix it.

**Comment by d-goldin** [comment]
@xvapx: Thanks for the review! While you are here, could you be interested in co-maintaining `pony`, which is a new dependency this tribler version brings in? (https://github.com/NixOS/nixpkgs/pull/74193)

**Review by xvapx — APPROVED** [review]

**Comment by xvapx** [comment]
@d-goldin: Thanks to you! I'm fine with co-maintaining pony, no problem.

**Comment by flokli** [comment]
Thanks :-)

@d-goldin, @xvapx: can one of you open a PR regarding the maintainorship of `pony`?

**Comment by xvapx** [comment]
@flokli: done in https://github.com/NixOS/nixpkgs/pull/74535, thanks!

**Comment by FlorianFranzen** [comment]
I think the update to Python3 was rather premature. I still experience regular crashes under Python3 that have not been fixed yet i.e. Tribler/tribler#5069. I would for now recommend to go back to version 7.3.2 under python2 before we hit the next stable release in March.

**Comment by ichorid** [comment]
Thanks for the bug report, guys! We're going to release 7.4 sometime next week. We deem it pretty stable after all the experimental releases. However, you should really wait until we're done with it, as one really important change regarding state/config directory handling is not implemented yet. Also, 7.5 is in the works and stable enough for testing now. It dumps Twisted in favour of Asyncio, adds Pathlib path handling, features complete package structure refactoring to adhere to Python standards, etc. Stay tuned!

**Comment by xvapx** [comment]
@FlorianFranzen Thanks for the report, I'll look into it and either fix it or downgrade to 7.3.2 before 20.03

**Pull request merged**
